packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol

```diff
@@ -913,12 +929,48 @@ contract FaultDisputeGame is Clone, ISemver {
         requiredBond_ = assumedBaseFee * requiredGas;
     }
 
+    function _tryUpdateAnchorState() internal {
+        try ANCHOR_STATE_REGISTRY.setAnchorState(IFaultDisputeGame(address(this))) { } catch { }
+    }
+
     /// @notice Claim the credit belonging to the recipient address.
+    /// @dev Reverts if the game isn't finalized, if the recipient has no credit to claim, or if the bond transfer
+    /// fails.
+    /// @dev If the game is finalized but no bond has been paid out yet, this method will determine the bond
+    /// distribution mode and also try to update the anchor state registry's anchor game.
     /// @param _recipient The owner and recipient of the credit.
     function claimCredit(address _recipient) external {
-        // Remove the credit from the recipient prior to performing the external call.
-        uint256 recipientCredit = credit[_recipient];
-        credit[_recipient] = 0;
+        if (bondDistributionMode == BondDistributionMode.UNDECIDED) {
+            // If the game is not finalized, bond claiming is not allowed; we defer a decision on distribution mode.
+            if (!ANCHOR_STATE_REGISTRY.isGameFinalized(IFaultDisputeGame(address(this)))) revert GameNotFinalized();
+
+            // If the game is finalized, update the anchor state and determine the bond distribution mode.
+            _tryUpdateAnchorState();
+
+            if (
```

Comment:
I think we should just be able to call `ANCHOR_STATE_REGISTRY.isValidGame(address(this))` right?

Reply:
No, because game validity excludes CHALLENGER_WINS states